executor/runtime/docker/docker.go

```diff
@@ -2969,33 +2971,33 @@ func (r *DockerRuntime) Kill(ctx context.Context, wasKilled bool) error { // nol
 		logger.G(ctx).Info("Stopping+Cleaning up containers because they finished or died")
 	}
 
-	if containerJSON, err := r.client.ContainerInspect(context.TODO(), r.c.ID()); docker.IsErrNotFound(err) {
+	if containerJSON, err := r.client.ContainerInspect(ctxWithTimeout, r.c.ID()); docker.IsErrNotFound(err) {
```

Comment:
Timing out here is good, but we should also retry.  I have observed at least one case in the wild where the HTTP request hung forever, but subsequent manual calls succeeded.  So transient failures are a real possibility I think.